File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        
        for(int i=0;i<s.size();i++)
        {
            char ch=s[i];
            if(ch=='(' or ch=='{' or ch=='[')
                st.push(ch);
            else
            {
                if(!st.empty())
                {
                    char top=st.top();
                    if((ch==')' and top=='(') or (ch=='}' and top=='{') or (ch==']' and top=='['))
                        st.pop();
                    else
                        return false;
                }
                else
                    return false;
            }
        }
        
        if(st.empty())
            return true;
        return false;
    }
};

File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
         if(s.length()==1){
            return false;
        }
        Stack<Character> st=new Stack<>();
        
        for(int i=0;i<s.length();i++){
            char c=s.charAt(i);
            if(c=='(' || c=='{' || c=='['){
                st.push(c);
            }
            else{
                char temp;
                if(st.size()!=0){
                    temp=st.pop();
                }
                else{
                    return false;
                }
                if((temp=='(' && c==')') || (temp=='[' && c==']') || (temp=='{' && c=='}')){
                    continue;
                }
                else{
                    return false;
                }
            }
            
        }
        if(st.size()!=0){
            return false;
        }
        return true;
    }
}